is comment or not/comment or not_0.cpp

```diff
@@ -0,0 +1,100 @@
+//Write a program to identify whether a given line is a comment or not
+
+#include<bits/stdc++.h>
+using namespace std;
+
+//Remove comments from string s
+void RemoveComments(string _s)
+{
+    //split s into code and comments
+    string code;
+    string comment;
+
+    //determines prev char to determine next char
+    enum CommentType
+    {
+        None,
+        slash,
+        SingleLine,
+        MultiLine
+    };
+    CommentType Current = CommentType::None;
+    
+    //loop through the file/string.
+    for (auto ch = _s.begin(); ch != _s.end(); ch++)
+    {
+        if(Current != CommentType::SingleLine && Current != CommentType::MultiLine)
+        {
+            //if not in a comment, and encounter a '/', change Current to slash as it could be a new comment
+            if(*ch == '/')
+            {
+                //if Current was slash, new / starts a single line comment
+                if(Current == CommentType::slash)
+                {
+                    Current = CommentType::SingleLine;
+                    code.pop_back();
+                }
+                else
+                    code += *ch;
+                if(Current == CommentType::None)
+                {
+                    Current = CommentType::slash;
+                }
+            }
+            //detect if it could be start of new multi-line comment
+            else if(*ch == '*')
+            {
+                //if it was slash, new * starts a multi line comment
+                if(Current == CommentType::slash)
+                {
+                    Current = CommentType::MultiLine;
+                    code.pop_back();
+                }
+                else
+                    code += *ch;
+            }
+            else
+                code += *ch;
+        }
+        //if already in a single line comment '\n' can break the comment
+        else if (Current == CommentType::SingleLine)
+        {
+            if(*ch == '\n')
+            {
+                Current = CommentType::None;
+            }
+            comment += *ch;
+        }
+        else // if (Current == CommentType::MultiLine)
+        {
+            //encountered * could be the end of multi-line comment if / follows
+            if(*ch == '*')
+            {
+                comment += *ch;
+                
+                ch++;
```

Comment:
This right here breaks the flow. everything checked char by char but here it is incrementing ch. 
you could have added an enum like "multilineEnd" or something to coverup.

Reply:
As i suspeected, it is resulting as a bug. you can not close a multiline comment with **/

Reply:
Fixed